BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift

```diff
@@ -453,41 +473,46 @@ extension ConvivaAnalytics: BitmovinPlayerListenerDelegate {
             return
         }
 
-        playerStateManager.setSeekEnd!(-1)
+        videoAnalytics?.reportPlaybackMetric(CIS_SSDK_PLAYBACK_METRIC_SEEK_ENDED, value: -1)
     }
 
     #if !os(tvOS)
+
     // MARK: - Ad events
+
     func onAdStarted(_ event: AdStartedEvent) {
         let adPosition: AdPosition = AdEventUtil.parseAdPosition(event: event, contentDuration: player.duration)
-        client.adStart(sessionKey, adStream: .ADSTREAM_SEPARATE, adPlayer: .ADPLAYER_CONTENT, adPosition: adPosition)
+
+        var adInfo = [String: Any]()
+        adInfo["c3.ad.position"] = adPosition
+
+        adAnalytics?.reportAdStarted(adInfo)
     }
 
     func onAdFinished() {
-        client.adEnd(sessionKey)
+        adAnalytics?.reportAdEnded()
     }
 
     func onAdSkipped(_ event: AdSkippedEvent) {
         customEvent(event: event)
-        client.adEnd(sessionKey)
+        adAnalytics?.reportAdSkipped()
     }
 
     func onAdError(_ event: AdErrorEvent) {
         customEvent(event: event)
-        client.adEnd(sessionKey)
+        adAnalytics?.reportAdError(event.message, severity: .ERROR_WARNING)
     }
 
     func onAdBreakStarted(_ event: AdBreakStartedEvent) {
-        customEvent(event: event)
-        client.detachPlayer(sessionKey)
+        videoAnalytics?.reportAdBreakStarted(AdPlayer.ADPLAYER_SEPARATE,
+                                             adType: AdTechnology.CLIENT_SIDE,
```

Comment:
Can we make this configurable (but with `CLIENT_SIDE` as default)? There are use cases where we have server-side ad insertion but the player emits the ad events in the same way as if it would be client-side.